### ISSUE TITLE
How to prevent validations on page load?

### ISSUE THREAD

**Issue by alex-dow** [monterail/vuelidate#403]
I'm obligated to do server-side validation for each form field.

My component, once mounted, loads data from the server, populates the form fields. But by doing so, vuelidate then validates each field against the server.

Thus, for my particular form where there are four elements, I make four separate requests to the server.

Is there any way to prevent this from happening?

**Comment by alex-dow** [comment]
https://jsfiddle.net/y70p6cw5/

This example better highlights my issue.

I don't expect ANY validation to be taking place, but instead, validation happens TWICE. I don't know how to work around this.

**Comment by alex-dow** [comment]
I am using vuelidate 0.7.4

**Comment by alex-dow** [comment]
https://jsfiddle.net/y70p6cw5/1/

In this second example, I tried to use a loading flag to no avail.

**Comment by venticco** [comment]
Validations works like computed properties so they are fired up once component is created. I'm not sure exactly if I understand this problem - could you describe what are expected scenario steps with your second fiddle example and highlight what part of current behavior is wrong?

**Comment by alex-dow** [comment]
Certainly:

1. Data comes from a server
2. Data from server is applied to data model of component
3. This triggers validation for each field in the model
4. Each validation requires another server call

What I want

1. Data comes from a server
2. Data from server is applied to data model of component
3. No validation happens at this point in time
4. Validation continues to behave normally when user makes changes

In case it matters, we are using only custom form components.

Now I can't see how I can make this work, since when a component is loaded, it's data model always starts off as being empty, and thus when data comes from the server and populates the data model, it always triggers whatever change/input events that then triggers vuelidate.

In my attempt to mitigate this, I used that "loaded" flag that is set when when the data is loaded from the server, and in my asyncValidation validator, I check to see if the loaded flag is false and if so, do nothing, otherwise return the promise that makes the server request to validate the value. This loaded flag doesn't appear to do anything at all.

**Comment by kosanr1** [comment]
> In my attempt to mitigate this, I used that "loaded" flag that is set when when the data is loaded from the server, and in my asyncValidation validator, I check to see if the loaded flag is false and if so, do nothing, otherwise return the promise that makes the server request to validate the value. This loaded flag doesn't appear to do anything at all.

This `loaded` flag stops the execution of the validation. If you remove `this.loaded = true`, your request will never get triggered. So it works as expected. When you change this flag to `true`, validation will be checked again and it will start promise. If you want to wait for user action, then you should use `$dirty` flag with your `loaded` flag. https://jsfiddle.net/y70p6cw5/2/

**Comment by jean-moldovan** [comment]
Hello, @alex-dow, in my understanding you may want to use `@blur` event to touch form (make it dirty):
https://jsfiddle.net/y70p6cw5/8/

Hope this helps!

**Comment by Liioo** [comment]
@alex-dow any news on this?

**Comment by alex-dow** [comment]
Sorry for the late response. I had to start working on other things.

But I've since returned to this. Managing $dirty flag manually prevents validations from running on page load, but calling $touch() on a single form field runs validations across the entire form now, which is also not ideal since these are server side validations.

But my situation has become a little more complicated as well. Essentially, I have a form component and I give it a list of field names. The form component, based on this list, prepared the validations object so that there is a validation method per field. The fields are deeply nested.

I will try to put together a jsfiddle to better explain the situation, but I think it might warrant a new issue.

**Comment by alex-dow** [comment]
well

https://jsfiddle.net/x9utmdov/2/

This is more accurate to what I'm trying to do now. All validations are run on page load.

In similar code on my project (which is corporate, can't share that code ;) ), all validations are not run on page load, but they are all run the minute one field changes (event though I only call $touch() on a single field).

**Comment by alex-dow** [comment]
I suppose my confusion is that, I assume if $touch() is not called, validations won't run, but this doesn't appear to be the case. Even if I comment out the this.$set() bit in my fiddle above, the whole form is still validated on page load.

**Comment by kosanr1** [comment]
Look at this example https://jsfiddle.net/07cmjweb/
I added `$dirty` flag. So now validation check if page is loaded and also if input was touched.
Also text input emits `input` event so you should watch @input not @change
And I've changed :value to v-model. I hope it will help you :)

**Comment by alex-dow** [comment]
https://jsfiddle.net/j1Lbserw/

doing it this way means I'm not sending a server request with every letter entered, and validation works only on blur.

I will have to try this out on my actual code. Thanks all for the input.

**Comment by kosanr1** [comment]
That's true, you can use debounce function to avoid checking validation with every letter.
Good luck :)

**Comment by alex-dow** [comment]
hmm

https://jsfiddle.net/xr5azd8p/

In this updated fiddle,t he validation is still being run on page load. Granted, it won't ping the server in this example, but the fact that it still runs is a bit frustrating.

Is there a way to prevent that behaviour specifically?

**Comment by joshuachinemezu** [comment]
Had same issue, had to solve it with this hack ):

`export default {
  data() {
    return {
      email: "",
      password: "",
      validate: false
    };
  },
validations() {
    if (this.validate) {
      return {
        email: {
          email,
          required
        },
        password: {
          required
        }
      };
    } else {
      return {
        email: {},
        password: {}
      };
    }
  },
formSubmit() {
      this.validate = true;
      this.$v.$touch();
      if (!this.$v.$invalid) {
       this.login({ email: this.email, password: this.password });
      }
    }
};
`

Hope this helps );

**Comment by i-figueroa** [comment]
The best approach is to check if the value is dirty and valid. On load, the values are never dirty. You should then handle the value when it changes or on input and touch the validation property to trigger the the change of the dirty flag.
